batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -0,0 +1,213 @@
+            return Ok(None);
+        } else {
+            process_batch_inclusion_data(
+                msg,
+                &mut aligned_verification_data,
+                verification_data_commitments_rev,
+                &mut stream,
+                &mut verified_batch_merkle_roots,
+                num_responses.clone(),
+            )
+            .await?;
+
+            if *num_responses.lock().await == total_messages {
+                debug!("All messages responded. Closing connection...");
+                ws_write.lock().await.close().await?;
+                return Ok(Some(aligned_verification_data));
+            }
+        }
+    }
+
+    Ok(None)
+}
+
+async fn process_batch_inclusion_data<'s>(
+    msg: Message,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    stream: &mut BatchVerifiedEventStream<'s>,
+    verified_batch_merkle_roots: &mut HashSet<Vec<u8>>,
+    num_responses: Arc<Mutex<usize>>,
+) -> Result<(), SubmitError> {
+    let mut num_responses_lock = num_responses.lock().await;
+    *num_responses_lock += 1;
+
+    let data = msg.into_data();
+    match serde_json::from_slice::<ResponseMessage>(&data) {
+        Ok(ResponseMessage::BatchInclusionData(batch_inclusion_data)) => {
+            handle_batch_inclusion_data(
+                batch_inclusion_data,
+                aligned_verification_data,
+                verification_data_commitments_rev,
+                stream,
+                verified_batch_merkle_roots,
+            )
+            .await?;
+        }
+        Ok(ResponseMessage::ProtocolVersion(_)) => {
+            error!("Batcher responded with protocol version instead of batch inclusion data");
+        }
+        Err(e) => {
+            error!("Error while deserializing batcher response: {}", e);
+        }
+    }
+
+    Ok(())
+}
+
+async fn process_batch_inclusion_data_without_await(
+    msg: Message,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    num_responses: Arc<Mutex<usize>>,
+) -> Result<(), SubmitError> {
+    let mut num_responses_lock = num_responses.lock().await;
+    *num_responses_lock += 1;
+
+    let data = msg.into_data();
+    match serde_json::from_slice::<ResponseMessage>(&data) {
+        Ok(ResponseMessage::BatchInclusionData(batch_inclusion_data)) => {
+            let _ = handle_batch_inclusion_data_without_await(
+                batch_inclusion_data,
+                aligned_verification_data,
+                verification_data_commitments_rev,
+            );
+        }
+        Ok(ResponseMessage::ProtocolVersion(_)) => {
+            error!("Batcher responded with protocol version instead of batch inclusion data");
+        }
+        Err(e) => {
+            error!("Error while deserializing batcher response: {}", e);
+        }
```

Comment:
return error

Reply:
Solved in [cef4afb](https://github.com/yetanotherco/aligned_layer/pull/583/commits/cef4afb1e728da34f45728b42d6780e29a862585)